tracegen/internal/tracegen/config.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracegen
+
+import (
+	"flag"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+// Config describes the test scenario.
+type Config struct {
+	Workers  int
+	Traces   int
+	Marshal  bool
+	Pause    time.Duration
+	Duration time.Duration
+	Service  string
+
+	// OTLP config
+	Endpoint string
+	Insecure bool
+}
+
+// Flags registers config flags.
+func (c *Config) Flags(fs *flag.FlagSet) {
+	fs.IntVar(&c.Workers, "workers", 1, "Number of workers (goroutines) to run")
+	fs.IntVar(&c.Traces, "traces", 1, "Number of traces to generate in each worker (ignored if duration is provided")
+	fs.BoolVar(&c.Marshal, "marshal", false, "Whether to marshal trace context via HTTP headers")
+	fs.DurationVar(&c.Pause, "pause", time.Microsecond, "How long to pause before finishing trace")
+	fs.DurationVar(&c.Duration, "duration", 0, "For how long to run the test")
+	fs.StringVar(&c.Service, "service", "tracegen", "Service name to use")
+
+	// unfortunately, at this moment, the otel-go client doesn't support configuring OTLP via env vars
+	fs.StringVar(&c.Endpoint, "otlp-endpoint", "localhost:55680", "Target to which the exporter is going to send spans or metrics. This MAY be configured to include a path (e.g. example.com/v1/traces)")
+	fs.BoolVar(&c.Insecure, "otlp-insecure", false, "Whether to enable client transport security for the exporter's grpc or http connection")
+}
+
+// Run executes the test scenario.
+func Run(c *Config, logger logr.Logger) error {
+	if c.Duration > 0 {
+		c.Traces = 0
+	} else if c.Traces <= 0 {
+		return fmt.Errorf("either `traces` or `duration` must be greater than 0")
+	}
+
+	wg := sync.WaitGroup{}
+	var running uint32 = 1
```

Comment:
this one is never used to signal the workers to stop

Reply:
Isn't it used a few lines below?

```go
if c.Duration > 0 {
	time.Sleep(c.Duration)
	atomic.StoreUint32(&running, 0)
}
```